src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,53 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int CAPACITY = 10;
+    private static final double COEFFICIENT = 0.5;
+    private static final int ONE = 1;
```

Comment:
okay, we may not make constants out of 1 and 0 numbers, they are not considered magic

Reply:
done